Add mutability changes to Cadence documentation (#172)
onflow/docs: docs/cadence/language/access-control.md

}

resource OuterResource {

>     access(Map) let childResource: @SubResource

Author: I did not specifically add about resource typed fields, since this section already describes how entitlement mappings pass down the entitlements in general. Also, the newly added `Field and Index Access` (in reference.md) section describes how references and entitlements are received when accessing fields through a reference.

If we want to have a more granular/in-depth explanation on how entitlement mappings behave on each different use case (functions, references, and container-typed field) separately, then I guess that also makes sense and would be good to have it too. But probably unrelated to this change, so maybe we could do it in a separate PR.

}
```

attempting to map `(A | D)` through `M` will fail, since `A` should map to `(B, C)` and `D` should map to `E`, but these two outputs cannot be combined into a disjunctive set.

### Built-in Mutability Entitlements

A prominent use-case of entitlements is to control access to object based on mutability.
For example, in a struct/resource/contract, the author would want to control the access to certain fields to be read-only,
and while some fields to be mutable, etc.

In order to support this, Cadence hase built-in set of entitlements that can be used to access control base on mutability.
- `Insert`
- `Remove`
- `Mutate`

These are primarily used by built-in array and dictionary functions, but are also usable by any user to control access
in their own composite type definitions.

> While there is no such a thing as entitlement composition/inheritance, the `Mutate` entitlement is intended to be used

Reviewer: ```suggestion
While Cadence does not support entitlement composition or inheritance, the `Mutate` entitlement is intended to be used
```
